crates/storage/libmdbx-rs/src/txn_manager.rs

```diff
@@ -239,22 +196,30 @@ mod read_transactions {
                     // Iterate through active read transactions and abort those that's open for
                     // longer than `self.max_duration`.
                     for entry in self.active.iter() {
-                        let (ptr, start) = entry.pair();
+                        let (tx, start) = entry.value();
                         let duration = now - *start;
 
                         if duration > self.max_duration {
-                            let ptr = *ptr as *mut ffi::MDBX_txn;
-
-                            // Add the transaction to the list of aborted transactions, so further
-                            // usages report the correct error when the transaction is closed.
-                            self.add_aborted(ptr);
-
-                            // Abort the transaction
-                            let result = mdbx_result(unsafe { ffi::mdbx_txn_abort(ptr) });
-
-                            // Add the transaction to `aborted_active`. We can't remove it instantly
-                            // from the list of active transactions, because we iterate through it.
-                            aborted_active.push((ptr, duration, result.err()));
+                            let result = tx.txn_execute(|txn_ptr| {
+                                (
+                                    txn_ptr,
+                                    duration,
+                                    // Abort the transaction
```

Comment:
```suggestion
                                    // Reset the transaction
```

Reply:
explained in a comment https://github.com/paradigmxyz/reth/blob/4d7687352b8b1a87f3d89e306b8d9547575fdbb3/crates/storage/libmdbx-rs/src/txn_manager.rs#L207-L216

Reply:
it wouldn't be we (the monitor) accidentally aborting a new transaction in that case, it would be the user who still holds the ptr in some old instance of some user type, the ptr that mdbx has now assigned again via a new call to open a tx. the user may drop that old instance along with the ptr and then accidentally abort a transaction they didn't intend.

Reply:
personally I also prefer when the right naming is used so use reset if reset is what is happening under the hood. it helps experienced developers read (and like) the code, otherwise you get the classic moment of
<img width="345" alt="Screenshot 2024-02-28 at 18 35 53" src="https://github.com/paradigmxyz/reth/assets/58548332/9bc973e0-21b0-49e1-ae45-0d9c785b3d8d">

Reply:
we use the "abort" terminology throughout the `read-tx-timeouts` feature, and IMO it becomes more confusing if we rename it to "reset". I think the comment explaining why "Abort the transaction" actually does "Reset the transaction" is enough.

Reply:
yeah it would require changing in more places

Reply:
changed to "time out" everywhere